### PR TITLE
Update to extend support for MonsterAPI new LLM APIs

### PR DESCRIPTION
**Pull request by Vikasqblocks**
# Description

MonsterAPI has updated its API structure to openAI API routes and synchronous APIs.

Update integration module and example colabnotebook to use with new structure.

## Version Bump?

Did I bump the version in the `pyproject.toml` file of the package I am updating? (Except for the `llama-index-core` package)

- [x] Yes
- [ ] No

## Type of Change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Added new unit/integration tests
- [x] Added new notebook (that tests end-to-end)
- [x] I stared at the code and made sure it makes sense

## Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added Google Colab support for the newly added notebooks.
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

File: llama-index-integrations/llms/llama-index-llms-monsterapi/README.md [hunk 1]
```diff
@@ -1 +1,53 @@
...
+Examples:
+
+`pip install llama-index-llms-monsterapi`
+
+1.  MonsterAPI Private LLM Deployment use case
+
+    ````python
+    from llama_index.llms.monsterapi import MonsterLLM
+
+        llm = MonsterLLM(
```
Comment (anoopshrma) [inline llama-index-integrations/llms/llama-index-llms-monsterapi/README.md:31]: Small correction needed here: 
base_url -> api_base
monster_api_key -> api_key

**Review by anoopshrma — COMMENTED** [review]
Thank you for the contribution!  I have mentioned a small change also test cases needs to be updated as well

File: llama-index-integrations/llms/llama-index-llms-monsterapi/README.md (hunk 1 above)
Comment (logan-markewich) [reply llama-index-integrations/llms/llama-index-llms-monsterapi/README.md:31]: Good catch! This is basically the only thing I noticed as well

Comment (Vikasqblocks) [reply llama-index-integrations/llms/llama-index-llms-monsterapi/README.md:31]: Done, updated! Good catch!

Comment (Vikasqblocks) [reply llama-index-integrations/llms/llama-index-llms-monsterapi/README.md:31]: @logan-markewich @anoopshrma Can you guys look at the CI test failing and help me with it? we don't have monsterapi package dependency anymore. I updated the pyproject.toml accordingly not sure where the reference of monsterapi is coming in!

Comment (Vikasqblocks) [reply llama-index-integrations/llms/llama-index-llms-monsterapi/README.md:31]: @logan-markewich @anoopshrma Need a little help here!

**Review by logan-markewich — APPROVED** [review]

**Pull request merged**